python/benchmark/test_gen_data.py

```diff
@@ -257,13 +365,70 @@ def test_make_sparse_regression(
         total = row_num * col_num
 
         # If there is no random shuffled redundant cols, we can check the total density
-        density_num = float(density)
-        if redundant_cols == "0":
+        if redundant_cols == "0" and density_curve == "None":
+            if isinstance(density, List):
+                density_num = sum([float(d) for d in density]) / len(density)
+            else:
+                density_num = float(density)
+
             assert (
                 count > total * density_num * 0.95
                 and count < total * density_num * 1.05
             )
 
+        # If no shuffle, test to see if the chunk density is as specified/curved
+        n_chunks_num = int(n_chunks)
+        if shuffle == "False":
+            orig_cols = col_num - int(redundant_cols)
+            num_partitions = 3
+
+            if isinstance(density, List):
+                density_num = float(density[0])
+            else:
+                density_num = float(density)
+
+            if density_curve == "Linear":
+                density_values = np.linspace(
+                    num_partitions / row_num, density_num, n_chunks_num
+                )
+                density_values *= n_chunks_num * density_num / sum(density_values)
+            else:
+                density_values = np.logspace(
+                    np.log10(num_partitions / row_num),
+                    np.log10(density_num),
+                    n_chunks_num,
+                )
+                density_values *= n_chunks_num * density_num / sum(density_values)
+
+            col_per_chunk = np.full(n_chunks_num, orig_cols // n_chunks_num)
+            col_per_chunk[: (orig_cols % n_chunks_num)] += 1
+            col_per_chunk = np.cumsum(col_per_chunk)
+
+            density_idx = 0
+            dense_count = 0
+            for idx, col in enumerate(X_np):
+                # Ignore the redundant columns
+                if idx >= orig_cols:
+                    break
+
+                if density_values[density_idx] == idx:
```

Comment:
not sure but it seems the if statement is always False? density_values[density_idx] is a float type, while idx is an int type?